include/sound/sof/control.h

```diff
@@ -149,7 +149,7 @@ struct sof_ipc_comp_event {
 	/* control data - add new types if needed */
 	union {
 		/* data can be used by binary controls */
-		struct sof_abi_hdr data[0];
+		DECLARE_FLEX_ARRAY(struct sof_abi_hdr, data);
```

Comment:
Sparse stops earlier, but this will not help either...
```
sound/soc/sof/ipc3-topology.c: note: in included file (through sound/soc/sof/sof-audio.h):
include/sound/sof/control.h:124:40: warning: array of flexible structures
```